### DISCUSSION TITLE
Unhandled Rejection (TypeError): Failed to fetch

### DISCUSSION THREAD

**Discussion by kiloppertry** [gadicc/yahoo-finance2#456] [Q&A]
import React from "react";
import yahooFinance from 'yahoo-finance2';

const Stocks = () => {
  const result =  yahooFinance.historical('TSLA', { period1: '2022-02-01',period2:'today'});
  console.log(result);

  return <div>Stocks</div>;
};

export default Stocks;

failed:

 3 | import yahooFinance from 'yahoo-finance2';
  4 | 
  5 | const Stocks = () => {
> 6 |   const result =  yahooFinan
![屏幕截图 2022-04-27 154849](https://user-images.githubusercontent.com/39750125/165468563-c6fed8e4-f2e5-4710-a279-b265771d5f82.jpg)
ce.historical('TSLA', { period1: '2022-02-01',period2:'today'});
  7 |   console.log(result);
  8 | 
  9 |   return <div>Stocks</div>;

**Comment by gadicc** [comment]
Hey @kiloppertry 

This is failing because you're passing `period2` a value of `"today"` which is not a valid value.  Per the [historical docs](https://github.com/gadicc/node-yahoo-finance2/blob/devel/docs/modules/historical.md), this should be a `date` (or a string value that can be passed to `new Date(str)`).  Of course, it mentions also that the default value is today, so you should just leave out `period2` entirely to get all history up until the current day.  i.e.

```js
yahooFinance.historical('TSLA', { period1: '2022-02-01',period2:'today'});  // won't work
yahooFinance.historical('TSLA', { period1: '2022-02-01' ); // will work
```

We should probably throw a clearer error if pass a string that can't be turned into a date, and I'll track that in #457.

**Reply by kiloppertry** [reply]
thank you for your answer ,but
![屏幕截图 2022-04-27 222414](https://user-images.githubusercontent.com/39750125/165541210-396f1f43-f69c-460b-b608-19cfd750887e.jpg)
![屏幕截图 2022-04-27 154849](https://user-images.githubusercontent.com/39750125/165541227-58a651d4-762c-436a-8651-b7650ca35ed3.jpg)
still failed

**Reply by gadicc** [reply]
Oh, oops, the `period2` was the first thing I noticed but looks like there are a bunch of other errors too.   Unfortunately I can't see from what you pasted the full error you're getting.  Anyways, a few other things:

* Don't forget that it's an async function, so you need to either `await` it or `.then()` it and handle that appropriately in your code.
* Are you running this in the browser?  Hope you read the [note in the README](https://github.com/gadicc/node-yahoo-finance2#using-in-the-browser-tldr-possible-but-unsupported) about this (blocked by CORS).
* Don't forget to `try...catch` (async) or `.catch(e) {...}` (promise) - it will help you get a better error message too.

Hope that helps!

**Reply by kiloppertry** [reply]
thank you , I am running in the browser . so its destiney is to fail.

**Reply by gadicc** [reply]
Ah sorry to hear that, @kiloppertry.  Yes, unfortunately there's no simple way to run this in the browser as Yahoo blocks those requests.  I hope you'll find an alternative method that be easier and work better.  Good luck!
